From: Sahil Chandna <chandna.sahil@gmail.com>
Subject: mm: pagewalk: simplify hugepage boundary calculation
Date: Fri, 28 Nov 2025 12:31:42 +0530

Patch series "mm: Coccinelle-driven cleanups across memory management
code", v4.

This patch series contains three minor cleanups identified by Coccinelle
scripts under scripts/coccinelle/misc/:

  * Use ALIGN() to compute the next hugepage boundary instead of the
    open-coded expression and use min() macro to improve readability.
  * Use the %pe printk format specifier for PTR_ERR() reporting
    in vmscan.c and zswap.c
Found using:
  make coccicheck MODE=report M=mm/


This patch (of 3):

Replace hugepage boundary computation with ALIGN() helper instead of an
open coded expression.  This helps to improves code readability.

This was flagged by Coccinelle (misc/minmax.cocci) as an opportunity to
use min(), after which the boundary computation was updated following
review suggestions.

Found by: make coccicheck MODE=report M=mm/
No functional change intended.

Link: https://lkml.kernel.org/r/cover.1764312627.git.chandna.sahil@gmail.com
Link: https://lkml.kernel.org/r/35ab82f052d248c0047c10bf01853e37e4f9b4e4.1764312627.git.chandna.sahil@gmail.com
Signed-off-by: Sahil Chandna <chandna.sahil@gmail.com>
Acked-by: David Hildenbrand (Red Hat) <david@kernel.org>
Suggested-by: David Hildenbrand (Red Hat) <david@kernel.org>
Suggested-by: Matthew Wilcox <willy@infradead.org>
Suggested-by: David Laight <david.laight.linux@gmail.com>
Cc: Axel Rasmussen <axelrasmussen@google.com>
Cc: Chengming Zhou <chengming.zhou@linux.dev>
Cc: Johannes Weiner <hannes@cmpxchg.org>
Cc: Liam Howlett <liam.howlett@oracle.com>
Cc: Lorenzo Stoakes <lorenzo.stoakes@oracle.com>
Cc: Michal Hocko <mhocko@suse.com>
Cc: Mike Rapoport <rppt@kernel.org>
Cc: Nhat Pham <nphamcs@gmail.com>
Cc: Qi Zheng <zhengqi.arch@bytedance.com>
Cc: SeongJae Park <sj@kernel.org>
Cc: Shakeel Butt <shakeel.butt@linux.dev>
Cc: Suren Baghdasaryan <surenb@google.com>
Cc: Vlastimil Babka <vbabka@suse.cz>
Cc: Wei Xu <weixugc@google.com>
Cc: Yosry Ahmed <yosry.ahmed@linux.dev>
Cc: Yuanchu Xie <yuanchu@google.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---


--- a/mm/pagewalk.c~mm-pagewalk-simplify-hugepage-boundary-calculation
+++ a/mm/pagewalk.c
@@ -312,8 +312,7 @@ static int walk_pgd_range(unsigned long
 static unsigned long hugetlb_entry_end(struct hstate *h, unsigned long addr,
 				       unsigned long end)
 {
-	unsigned long boundary = (addr & huge_page_mask(h)) + huge_page_size(h);
-	return boundary < end ? boundary : end;
+	return min(ALIGN(addr, huge_page_size(h)), end);
 }
 
 static int walk_hugetlb_range(unsigned long addr, unsigned long end,
_
